### DISCUSSION TITLE
Configurable MQTT Topic

### DISCUSSION THREAD

**Discussion by Mr-Eskildsen** [HASwitchPlate/openHASP#401] [Ideas]
The MQTT topic is always hasp/<platename/
For some special cases it would be nice to It would be nice to make this customizeable. 
For instance I might want to group some plates on a topic where I want to limit access on subscribing to the topic.

**Comment by fvanroie** [comment]
Can you please describe in more detail your concrete need, what topics you require and why?
This is a convention that's been in place since HASwitchPlate and there hasn't been a need so far.
I would like to understand why `hasp/<platename>/` is not sufficient for your use case.

**Reply by Mr-Eskildsen** [reply]
The request/idea is to make "hasp/" configurable. It is more because I would like to group plates, eg ifi I have one plate that i use for a diy alarm it would be nice to group all topics in eg. "my-alarm/plate/.....".

**Comment by Humancell** [comment]
Yes!  I really want this also.  I want to have multiple different "customers" sharing the same MQTT server.  Using the Dynamic Security Plugin we'll be managing the users and the Access Control Lists (ACLs) for each user.

We also want to group the users switch plates under a configurable topic something like `hasp/<account_id>/<platename>` so that we can assign the ACL to `hasp/<account_id>` for the user who owns the plates under that account.

**Comment by fvanroie** [comment]
@Mr-Eskildsen and @Humancell in  `Configurable MQTT Topic` #428  the MQTT Topics have now been made configurable.
Does this solve your problem with the fixed MQTT topcics?

**Reply by Mr-Eskildsen** [reply]
As I read the code you have added a configurable groupname to the topic, thus it is possible to configure the topic.

**Comment by chbarg** [comment]
Hello,

Does anybody know if its possible to setup OpenHASP to directly subscribe to a topic instead of having Home Assistant feed the information?

I would like OpenHASP to display sensor values reported by zigbee2mqtt, independently from HA.

I have a display running but I need some software that subscribes to (for example) zigbee2mqtt/m5p-co2-01/sensor/m5p-co2-01_co2_temperature/state  and creates a new message like 'hasp/plate01/command/p1b3.text' -m  "25.4 C"

In my dreams, a OpenHASP object would directly subscribe to zigbee2mqtt/m5p-co2-01/sensor/m5p-co2-01_co2_temperature/state.

Thank you.
